File: leetcode/go/problem139.go
	func wordBreak(s string, wordDict []string) bool {
		dp := make([]bool, len(s)+1)
		dp[len(s)] = true
		for i := len(s) - 1; i >= 0; i-- {
			for _, v := range wordDict {
				if len(v)+i <= len(s) && s[i:i+len(v)] == v {
					dp[i] = dp[i+len(v)]
				}
				if dp[i] {
					break
				}
			}
		}
		return dp[0]
	}

File: leetcode/python/problem139.py
 class Solution:
    def wordBreak(self, s: str, wordDict: List[str]) -> bool:
        m = {}
        def f(s):
            if s in wordDict: return True
            if s in m: return m[s]
            if len(s) <= 1: return False
            res = False
            for i in range(1,len(s)):
                if res: break
                res = f(s[0:i]) and f(s[i:])
            m[s] = res
            return res
        return f(s)
